pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -280,188 +114,164 @@ var (
+	servicePrincipalFileContent := base64.StdEncoding.EncodeToString([]byte("msi"))
+	nBCB.GetNodeBootstrapConfig().AuthConfig.ServicePrincipalId = servicePrincipalClientID
+	nBCB.GetNodeBootstrapConfig().AuthConfig.ServicePrincipalSecret = servicePrincipalFileContent
+	nBCB.GetNodeBootstrapConfig().AuthConfig.AssignedIdentityId = a.UserAssignedIdentityID
+	nBCB.GetNodeBootstrapConfig().NetworkConfig.NetworkPlugin = nbcontractv1.GetNetworkPluginType(a.NetworkPlugin)
+	nBCB.GetNodeBootstrapConfig().NetworkConfig.NetworkPolicy = nbcontractv1.GetNetworkPolicyType(a.NetworkPolicy)
+	nBCB.GetNodeBootstrapConfig().KubernetesVersion = a.KubernetesVersion
+
+	nBCB.GetNodeBootstrapConfig().KubeBinaryConfig.KubeBinaryUrl = kubeBinaryURL(a.KubernetesVersion, a.Arch)
+	nBCB.GetNodeBootstrapConfig().NetworkConfig.VnetCniPluginsUrl = fmt.Sprintf("%s/azure-cni/v1.4.32/binaries/azure-vnet-cni-linux-%s-v1.4.32.tgz", globalAKSMirror, a.Arch)
+	nBCB.GetNodeBootstrapConfig().NetworkConfig.CniPluginsUrl = fmt.Sprintf("%s/cni-plugins/v1.1.1/binaries/cni-plugins-linux-%s-v1.1.1.tgz", globalAKSMirror, a.Arch)
 
-	nbv.NetworkPlugin = a.NetworkPlugin
-	nbv.NetworkPolicy = a.NetworkPolicy
-	nbv.KubernetesVersion = a.KubernetesVersion
+	// calculated values
+	nBCB.GetNodeBootstrapConfig().ClusterConfig.ClusterNetworkConfig.SecurityGroupName = fmt.Sprintf("aks-agentpool-%s-nsg", a.ClusterID)
+	nBCB.GetNodeBootstrapConfig().ClusterConfig.ClusterNetworkConfig.VnetName = fmt.Sprintf("aks-vnet-%s", a.ClusterID)
+	nBCB.GetNodeBootstrapConfig().ClusterConfig.ClusterNetworkConfig.RouteTable = fmt.Sprintf("aks-agentpool-%s-routetable", a.ClusterID)
 
-	nbv.KubeBinaryURL = kubeBinaryURL(a.KubernetesVersion, a.Arch)
-	nbv.VNETCNILinuxPluginsURL = fmt.Sprintf("%s/azure-cni/v1.4.32/binaries/azure-vnet-cni-linux-%s-v1.4.32.tgz", globalAKSMirror, a.Arch)
-	nbv.CNIPluginsURL = fmt.Sprintf("%s/cni-plugins/v1.1.1/binaries/cni-plugins-linux-%s-v1.1.1.tgz", globalAKSMirror, a.Arch)
+	nBCB.GetNodeBootstrapConfig().VmSize = a.VMSize
 
-	// calculated values
-	nbv.EnsureNoDupePromiscuousBridge = nbv.NeedsContainerd && nbv.NetworkPlugin == "kubenet" && nbv.NetworkPolicy != "calico"
-	nbv.NetworkSecurityGroup = fmt.Sprintf("aks-agentpool-%s-nsg", a.ClusterID)
-	nbv.RouteTable = fmt.Sprintf("aks-agentpool-%s-routetable", a.ClusterID)
-
-	if a.GPUNode {
-		nbv.GPUNode = true
-		nbv.ConfigGPUDriverIfNeeded = true
-		nbv.GPUDriverVersion = a.GPUDriverVersion
-		nbv.GPUImageSHA = a.GPUImageSHA
+	if agentbakercommon.IsNvidiaEnabledSKU(nBCB.GetNodeBootstrapConfig().VmSize) {
+		nBCB.GetNodeBootstrapConfig().GpuConfig.ConfigGpuDriver = true
 	}
-	nbv.NeedsCgroupV2 = true
+	nBCB.GetNodeBootstrapConfig().NeedsCgroupv2 = ptr.Bool(true)
 	// merge and stringify labels
 	kubeletLabels := lo.Assign(kubeletNodeLabelsBase, a.Labels)
 	getAgentbakerGeneratedLabels(a.ResourceGroup, kubeletLabels)
 
-	subnetParts, _ := utils.GetVnetSubnetIDComponents(a.SubnetID)
-	nbv.Subnet = subnetParts.SubnetName
```

Comment:
I think these assignments are added recently in https://github.com/Azure/karpenter-provider-azure/commit/02b76c639020b3cbfd1b3586215a7cdce580a1af. Let's include it in your changes too.

Reply:
Thanks for catching this. Updated